src/libcore/iter/adapters/zip.rs

```diff
@@ -142,6 +167,15 @@ where
         }
     }
 
+    #[inline]
```

Comment:
Unfortunately, this specialization is unsound, though it's not a pre-existing problem, and not specific to this PR. See [my comment here](https://github.com/rust-lang/rust/issues/67194#issuecomment-565595437) for a bit more context.

Reply:
Alright, so I think we should stop the work for #54054 until the problem(#67194) is resolved, right?

Reply:
Maybe -- there are specific patterns that are sound -- but you have to specialize not based on traits but based on concrete types. e.g., specializing for `vec::Iter` or something would be fine. But maybe there are so many types here that this is not practical?

Reply:
> but you have to specialize not based on traits but based on concrete types. e.g., specializing for vec::Iter or something would be fine.

Yeah, it'd be an alternative. But unfortunately, I have no time to find alternative implementation. If I find some time, I'll re-open or submit another PR. Thanks for the review, Niko!